### ISSUE TITLE
Ninja missing on SDL2 build on use in another library

### ISSUE THREAD

**Issue by tomvierjahn** [bincrafters/community#234]
### Description of Problem, Request, or Question
We are using SDL2 in another library via the other library's `conanfile.py`:
```Python
requires = (("sdl2/2.0.8@bincrafters/stable")
```
When building/installing the other library's dependencies via `conan install ../.. --build missing` from source, conan aborts with an error
```
ERROR: sdl2/2.0.8@bincrafters/stable: Error in build() method, line 149
	self.build_cmake()
while calling 'build_cmake', line 183
	cmake.configure(build_dir='build')
	ConanException: Error 256 while executing cd 'build' && cmake -G "Ninja" -DCMAKE_BUILD_TYPE="Release" -DCONAN_EXPORTED="1" -DCONAN_COMPILER="apple-clang" -DCONAN_COMPILER_VERSION="9.1" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/Users/tom/.conan/data/sdl2/2.0.8/bincrafters/stable/package/a863de2be5c29880f32adf086a97b766172889ef" -DCMAKE_POSITION_INDEPENDENT_CODE="True" -DSDL_SHARED="False" -DSDL_STATIC="True" -Wno-dev '/Users/tom/.conan/data/sdl2/2.0.8/bincrafters/stable/build/a863de2be5c29880f32adf086a97b766172889ef'
```
wich imho boils down to Ninja not being installed.

Prior to bincrafters/conan-sdl2@11e6fa2ba740a69518a85065f9a3823c76cb8f16 SDL2's conan recipe included Ninja via `build_requires`. Now, that these have moved to `build.py`, the `build_requires` are not available when building the package from source on usage. 

Bottom line: whenever there is no pre-built SDL2 binary package available, we will be unable to build our library without some hoop-jumping.

### Package Details (Include if Applicable)
* Package Name/Version: **sdl2/2.0.8@bincrafters/stable**
* Operating System: **Mac OS X**, **Centos**
* Compiler+version: **apple-clang-9.1**, **gcc-6**

### Steps to reproduce (Include if Applicable)
see above

**Comment by SSE4** [comment]
thanks for the feedback, I am removing explicit Ninja dependency from our packages. I'll keep you informed.

**Comment by tomvierjahn** [comment]
Thanks a lot!
